gateway/gateway-controller/cmd/controller/main_test.go

```
@@ -644,6 +647,19 @@ func TestGenerateAuthConfig(t *testing.T) {
 		assert.NotNil(t, authConfig.JWTConfig.PermissionMapping)
 		require.NotNil(t, authConfig.JWTConfig.Audience)
 		assert.Equal(t, []string{"gateway-controller"}, *authConfig.JWTConfig.Audience)
+		require.NotNil(t, authConfig.HTTPClient,
+			"JWKS fetching must use an SSRF-guarded client, never jwkset's unguarded default")
+		// The target is an IP literal, so this needs no real network access: the guarded
+		// dialer's resolution step returns the literal itself and the policy check rejects
+		// it before any connection is attempted (mirrors guarded_http_client_test.go's
+		// TestUpstreamFetchClientRefusesDeniedAddress in platform-api).
+		resp, err := authConfig.HTTPClient.Get("http://169.254.169.254/latest/meta-data/")
```

golangci-lint (2.12.2)

[error] (*net/http.Client).Get must not be called. use (*net/http.Client).Do(*http.Request)

(noctx)